### PR TITLE
South design

### PR DESCRIPTION
**Pull request by Mohs11**
Lastest update

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+    display: none;
+}
+
+.backgrounds {
+    list-style: none;
+    position: relative;
+    z-index: -1;
+    margin: 0;
+    padding: 0; 
+
```
Comment (fozcode) [inline css/style.css]: The backgrounds need to be hidden initially.  Roland has found that on a slow connection when the files take a long time to load you currently see all the backgrounds visible all at once.

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+]
+
+// pause boolean for background and title rotation
+let pause = false; 
+
+function updateHeading() {
+	let heading = document.getElementById('heading')
+	heading.innerHTML = headings[0];
+	heading.style.opacity = 1;
+	i = 1;
```
Comment (fozcode) [inline js/main.js]: `i` here is a global variable, was that intended? If it was it needs renaming as it clashes with `i` in other places.

Comment (fozcode) [reply js/main.js]: You can put:
`"use strict";`
at the top of the JS file to catch things like this.

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+			if (i > 0) {
+				backgrounds[i].style.opacity = 0;			
+					setTimeout(function(){
+						backgrounds[i].style.opacity = 1;
+					}, 500);
+					i = i - 1;
+				} else {				
+					setTimeout(function(){
+						i = 3;	
+						backgrounds[i].style.opacity = 1;
```
Comment (fozcode) [inline js/main.js]: I don't get this bit. `i=3` and `backgrounds[i]...` but the code then sets `backgrounds[3]...` anyway

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	let panel = e.target.dataset.panel;
+	let menu_tag = e.target.id;
+	let li_ele = document.getElementById(menu_tag);
+	let all_menu_li = document.getElementsByClassName('menu-option')
+	let to_be_opened = document.getElementById(panel);
+	let navigation = document.getElementById('navigation')
+	let allPanels = document.getElementsByClassName('side-panel');
+
+	// loop through the panels and remove the active class
+	for (let i = 0; i < allPanels.length; i++) {
```
Comment (fozcode) [inline js/main.js]: If you're using `i` for loops like this (that's fine, I do), let's not use `i` for also keeping track of which background and heading you're on.  Rename those others to something more meaningful like `bg_index` and `heading_index`.

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+			all_menu_li[i].classList.remove('li_active');
+		}
+		li_ele.classList.add('li_active');
+	}
+	let leftMainPanel = document.getElementById('leftMainPanel')
+	leftMainPanel.addEventListener("click", hideAll);
+}
+
+function hideAll(e) {
+	console.log('this was run')
```
Comment (fozcode) [inline js/main.js]: Need to remove these debugging logs when debugging is done

File: js/main.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+		icons.style.opacity = 1;
+	} else {
+		icons.style.opacity = 0;
+	}
+}
+
+function externalFeaturesNav(relate, link) {
+	let allTheFeatures = document.getElementsByClassName('ext-features-content');
+	let allTheMenuItems = document.getElementsByClassName('extra-features-ul-li');
+	for (i = 0; i < allTheFeatures.length; i++)	 {
```
Comment (fozcode) [inline js/main.js]: You're using `i` the global variable here. It needs a `let`. This is why the fading might be going weird when you click on certain things.

File: js/main.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+		allTheFeatures[i].style.display = 'none';
+		console.log(allTheFeatures[i])
+	}
+	for (i = 0; i < allTheMenuItems.length; i++)	 {
```
Comment (fozcode) [inline js/main.js]: As above, it's the wrong `i`

File: index.html [hunk 8]
```diff
@@ -216,236 +240,385 @@ <h1>
...
+
+                    <div id="external_features_two" class="ext-features-content" style="display: block;">
+                      <p class="extra-features-text-block">
+                        Imaging operations can be combined together and saved as templates. QIS supports the
+                        Python-Pillow
+                        library and ImageMagick package for its core image processing engine. Available operations
+                        include:
+                      </p>
+
+                      <div class="extra-features-list">
```
[the remark this answers is not in the document]
Comment (Mohs11) [reply index.html]: Thanks for that, I tried making it work before but I think I was using the wrong syntax! :)

File: js/main.js (hunk 2 above)
Comment (Mohs11) [reply js/main.js]: I was trying to figure out why this was happening. Thanks forgot the let. I've updated both variables to more suitable names now

**Pull request merged**